### ISSUE TITLE
Improve the behavior of the search bar in the Search Service

### ISSUE THREAD

**Issue by agpul** [cyfronet-fid/eosc-search-service#272]
The scope of this ticket is to make a full check of the search bar behavior and improve it. One issue that has been found during the live demo:
1.  Go to: https://search.eosc-portal.eu/
2. Type for eg: "amnesya" - see no result
3. Type: amnesia - see results
 As you can see if the user makes a mistake in typing the word there are no results.

**Comment by danielkryska** [comment]
https://solr.apache.org/guide/6_6/the-standard-query-parser.html#TheStandardQueryParser-FuzzySearches
https://solr.apache.org/guide/6_6/the-standard-query-parser.html#TheStandardQueryParser-ProximitySearches
https://solr.apache.org/guide/6_6/the-standard-query-parser.html#TheStandardQueryParser-ConstantScorewith_
https://solr.apache.org/guide/6_6/the-standard-query-parser.html#TheStandardQueryParser-TheBooleanOperator_

**Comment by danielkryska** [comment]
wild cards: + (single char), * (multiple chars)
long terms with white chars have been wrapped into "

**Comment by biometricsmw** [comment]
It seems that ~ is a good solution to show plenty of results. However the main problem here is that it is fuzzy search - so when we will try to observe "int" word (as the beginning of word "Introduction") we will also gain some results were "in" is observable. It is related to the methodology used for measuring the distance between the term and other words. When it comes to the possible solution to this problem is increasing the validity of the term with ^ - e.g., int^5 will tell solr that term int is important and should be searched as the first one. However, it seems that only combination of these two solutions can work appropriately. It is especially observable in the case of multiple terms - e.g., "int t" will show 0 results, "int t~" will show almost all results from the database, "int^5 t~" will reduce the number of the found records. However, we should also consider the importance of the terms when there are multiple of them. I would recommend to use the length of the term as "importance level". So our latest search will be "int^3 t^1 ~". 

To sum up:
- "~" is needed to run fuzzy search
- "^" is needed to assign importance level - to reduce the number of found records (importance level can be assigned on the basis of word length)

However, it is also interesting how "+" (required) will work. For now, it is removed from query string.

**Comment by biometricsmw** [comment]
Also ~ works well when it comes to typo in the words:
- Write q=persimtent -> no results
- Write q=persimtent~ -> the results are shown as well as they include those that are required by the user

**Comment by danielkryska** [comment]
Use cases for searching:

Software:
"title":"Monkeypox Knowledge Graph: A semantic resource embedding biological and chemical entities"
"title":"COVID-19 Knowledge Graph: A semantic resource embedding biological and chemical entities"
Other:
title:"ENVRI SP - Dashboard State of the Environment - Ocean Indicators"
title:"Mapping the photic zone of the Mediterranean Sea"

**Comment by danielkryska** [comment]
Possible solutions for search UX improvement:
- [ ] set additional search params contained in `backend/app/solr/operations.py`
- [ ] set better tokenization for searched fields 
- [ ] set the`_text_` field with all fields values copies by which we want to search
- [ ] set up a better query using fuzzy search sign `~` and/or wildcarding `*`, and any other special chars
